src/utils/framework.ts

```diff
@@ -75,6 +79,20 @@ interface LevelOutput {
     nodes?: LevelOutput[];
 }
 
+interface WidgetData {
+    data: {
+        value: string;
+    };
+}
+interface ConditionalAttribute {
+    selectedWidgetKey?: string;
+    [key: string]: WidgetData | string | undefined;
```

Comment:
Do we have both WidgetData and string on same mapping?

Reply:
i added string value as to comply selectedWidgetKey property with the second property specification

Reply:
No worries!